modules/json/src/lib/parse-ndjson-in-batches.ts

```diff
@@ -0,0 +1,33 @@
+import type {Batch} from '@loaders.gl/schema';
+import {TableBatchBuilder} from '@loaders.gl/schema';
+import {LoaderOptions, makeLineIterator, makeTextDecoderIterator} from '@loaders.gl/loader-utils';
+
+export default async function* parseNDJSONInBatches(
+  binaryAsyncIterator: AsyncIterable<ArrayBuffer> | Iterable<ArrayBuffer>,
+  options: LoaderOptions
+): AsyncIterable<Batch> {
+  const textIterator = makeTextDecoderIterator(binaryAsyncIterator);
+  const lineIterator = makeLineIterator(textIterator);
+
+  const schema = null;
+  const shape = 'row-table';
+  // @ts-ignore
+  const tableBatchBuilder = new TableBatchBuilder(schema, {
+    ...options,
+    shape
+  });
+
+  for await (const chunk of lineIterator) {
+    tableBatchBuilder.addObjectRow(JSON.parse(chunk));
```

Comment:
- I suppose the row could be an array, not just an object? Maybe that also fixes the `protected` CI error you are getting.
- Nit: I prefer breaking into two lines, a bit more readable
```suggestion
    const row = JSON.parse(chunk);
    tableBatchBuilder.addRow(row);
```

Reply:
It could be nice to track the line number (I think there is an iterator for that) and generate a better error. The error messages from JSON.parse are notoriously cryptic